src/gluonts/model/canonical/__init__.py

```diff
@@ -11,6 +11,16 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
-from ._estimator import CanonicalRNNEstimator
+import warnings
+
+from gluonts.mx.model.canonical import CanonicalRNNEstimator
+
+warnings.warn(
+    "The module gluonts.model.canonical has been moved to "
+    "gluonts.mx.model.canonical. In GluonTS v0.12 it will be no longer "
+    "possible to use the old path. Try to use 'from gluonts.mx import "
+    "CanonicalRNNEstimator'.",
+    FutureWarning,
```

Comment:
I think there is still something confusing about this.

We should think about what the canonical path of the models should be. In this case is `gluonts.mx.model.canonical` or `gluonts.mx` the thing that should be remembered?

I like the idea of exposing the models directly in `mx` and ideally people don't need to know that the `model` sub-package exists.

@lostella thoughts?